### PR TITLE
Storages: move ann_query_info to PushDownExecutor 

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### What is changed and how it works?

```commit-message
1. move `ann_query_info` to `PushDownFilter`
2. rename `PushDownFilter` to `PushDownExecutor`
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/KVStore/tests/gtest_kvstore_fast_add_peer.cpp [hunk 1]
```diff
@@ -596,7 +596,14 @@ try
         return genFastAddPeerResFail(FastAddPeerStatus::NoSuitable);
     });
     // Will generate and persist some information in local ps, which will not be uploaded.
-    FastAddPeerImplWrite(global_context.getTMTContext(), proxy_helper.get(), region_id, 2333, std::move(mock_data), 0);
+    auto mock_data_cp = mock_data;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/tests/gtest_kvstore_fast_add_peer.cpp:599]: why we need a copy here?

File: dbms/src/Storages/DeltaMerge/Segment.h [hunk 2]
```diff
@@ -751,13 +751,22 @@ class Segment
         const DMFilePackFilterResults & pack_filter_results,
         UInt64 start_ts,
         size_t expected_block_size);
-    SkippableBlockInputStreamPtr getConcatSkippableBlockInputStream(
+    std::tuple<SkippableBlockInputStreamPtr, bool> getConcatVectorIndexBlockInputStream(
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/Segment.h:754]: Add comment about the returing `bool`

File: dbms/src/Storages/KVStore/tests/gtest_kvstore_fast_add_peer.cpp (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/KVStore/tests/gtest_kvstore_fast_add_peer.cpp:599]:
```
    FastAddPeerImplWrite(global_context.getTMTContext(), proxy_helper.get(), region_id, 2333, std::move(mock_data), 0);
     dumpCheckpoint();
     FastAddPeerImplWrite(global_context.getTMTContext(), proxy_helper.get(), region_id, 2333, std::move(mock_data), 0);
```

`mock_data` be moved twice.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**